test/lint/commit-script-check.sh

```diff
@@ -31,7 +31,10 @@ for commit in $(git rev-list --reverse $1); do
         else
             echo "Running script for: $commit"
             echo "$SCRIPT"
-            (eval "$SCRIPT")
+            if ! (eval "$SCRIPT") && (echo "$SCRIPT" | grep --quiet "sed"); then
+                echo "Warning: do not use BSD sed syntax"
+                echo "Example: sed -i '' 's/foo/bar/' filename (Bad)"
```

Comment:
I would have expected a generic warning that the command failed, maybe something more like:

```sh
if ! (eval "$SCRIPT"); then
    echo "Warning: Script failed (non-zero exit code)"
    if echo "$SCRIPT" | grep -q "sed"; then
        echo "Suggestion: did you use BSD sed syntax?"
        echo "Example: sed -i '' 's/foo/bar/' filename (Bad)"
    fi
fi
```

Reply:
sed doesn't have an exit code for "error", so checking the exit code wouldn't work either way.


```
$ sed -i -e 's/FOOBAR/foobar/g' ./src/init.cpp && echo $?
0

Reply:
"sed -i '' -e 's/FOOBAR/foobar/g' ./src/init.cpp && echo $?" results in no output for me since the sed command fails and the "echo $?" command is not executed but "sed -i -e 's/FOOBAR/foobar/g' ./src/init.cpp" does in fact work. I think this is because "sed -i -e 's/FOOBAR/foobar/g' ./src/init.cpp" is a legitimate sed command, it just doesn't cause any changes since FOOBAR is not present in the file.

Reply:
In the OpenBSD 6.9 man pages for sed, it states EXIT STATUS The sed utility exits 0 on success, and >0 if an error occurs.

Reply:
failing to replace is not an error

Reply:
The line "git --no-pager diff --exit-code $commit && echo "OK" || (echo "Failed"; false) || RET=1" should check if the script failed to change anything, do you want to check each command and error if if has no change?

Reply:
Do you want to not check for non zero exit codes but only check if the output of the script is the same as the commit?